### ISSUE TITLE
Car parts are turning all green when falling off

### ISSUE THREAD

**Issue by MOnkeMan5009** [aap/skygfx#36]
in short, tearing down the bumper or hood of a car makes it turn green instead of whats it supposed to

PC Skygfx : 
![Grand Theft Auto  San Andreas Screenshot 2024 11 08 - 14 28 22 17](https://github.com/user-attachments/assets/cb4d0963-cc5b-45fc-8081-c88790c954d9)

PS2 Skygfx : 
![Grand Theft Auto  San Andreas Screenshot 2024 11 08 - 14 28 19 09](https://github.com/user-attachments/assets/97639909-83f0-496d-a8a5-080843e22f02)

![Grand Theft Auto  San Andreas Screenshot 2024 11 08 - 14 20 27 27](https://github.com/user-attachments/assets/1e5c3330-bf41-492f-9045-049fb8a0edf4)

another example of ps2 : 

and here is PC with no problem :
![Grand Theft Auto  San Andreas Screenshot 2024 11 08 - 14 27 44 13](https://github.com/user-attachments/assets/a37f93af-d6ea-40a2-a206-01903fc3171d)

I have latest silentpatch and skygfx

**Comment by Lethality5001** [comment]
@aap

**Comment by aap** [comment]
This was not a problem in the past, but perhaps it's related to the new version of silentpatch?

**Comment by Lethality5001** [comment]
> This was not a problem in the past, but perhaps it's related to the new version of silentpatch?

i remember that this was not a problem in the past, im not sure if its a silentpatch problem, because Pc skygfx is fine. i even tried a skygfx asi from 2022 as a test and same thing . i have no idea.
Can that silentpatch guy see this too? I aint the only one with this

**Comment by Lethality5001** [comment]
> yes, we have talked about it on discord and he's investigating it. so far it's difficult because he cannot reproduce it.

there seems to be other posts related to this on reddit too. some people said its skygfx fault

**Comment by MOnkeMan5009** [comment]
> yes, we have talked about it on discord and he's investigating it. so far it's difficult because he cannot reproduce it.

Can you send some older silentpatch file to test if thats the case? many say its skygfx yeah. Also, maybe give me ur personal discord? could test further. im pretty good at spotting imperfections lol

**Comment by MOnkeMan5009** [comment]
FOUND The problem. its silentpatch. took an older version of it. no problems. Look here recording, even switching GFX at the end for you to see : 

https://github.com/user-attachments/assets/55237e17-a43c-4bd8-970e-c512a7e9b98a

**Comment by CookiePLMonster** [comment]
> FOUND The problem. its silentpatch. took an older version of it. no problems. Look here recording, even switching GFX at the end for you to see :
> 
>  8mb.video-zuY-HlL7IYGj.mp4

Could you share your exact config file? I am using skygfx myself but never encountered this

**Comment by MOnkeMan5009** [comment]
i can't damn post the entire file here, so i guess zip with link it is : 
https://www.mediafire.com/file/d5xcq82m04qisfa/skygfx_configs.zip/file
(Also, im using skygfx 4.2 extended a or whatever it was called, apparently b got rid of some rain reflections as one guy said in another issue ticket )
But the configs didnt seem to matter much, PC worked with your latest silentpatch but the rest didn't. using an older silentpatch from like 2020 and now it works fine

**Comment by CookiePLMonster** [comment]
> (Also, im using skygfx 4.2 extended a or whatever it was called, apparently b got rid of some rain reflections as one guy said in another issue ticket )

With aap's SkyGfx, I cannot reproduce this issue on any of your configs.

![image](https://github.com/user-attachments/assets/c2336364-a991-4ce4-afc5-1fe76aafa894)

It might be something Extended introduced, but I can't know what.

**Comment by MOnkeMan5009** [comment]
> > (Also, im using skygfx 4.2 extended a or whatever it was called, apparently b got rid of some rain reflections as one guy said in another issue ticket )
> 
> With aap's SkyGfx, I cannot reproduce this issue on any of your configs.
> 
> ![image](https://private-user-images.githubusercontent.com/7947461/384692702-c2336364-a991-4ce4-afc5-1fe76aafa894.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyNDU3NjEsIm5iZiI6MTczMTI0NTQ2MSwicGF0aCI6Ii83OTQ3NDYxLzM4NDY5MjcwMi1jMjMzNjM2NC1hOTkxLTRjZTQtYWZjNS0xZmU3NmFhZmE4OTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTBUMTMzMTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzIyODg2YWMxYjI5ZDUzMTg2YjIzNGU1N2Q5MzkzNDcyM2ZmY2IyM2VhMGRjYTAxNWZlODYzYmNkNGZhZWIxYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GCjq7Xiv-eKu4Tb2AxF92On1Ak7VfyijWr-XIn3zkPk)
> 
> It might be something Extended introduced, but I can't know what.

perhaps... i seen you comment on some reddit post too related to this. but seems like an older version of silentpatch fixed it for me. Maybe something related to that? or the skygfx mod doesnt fully properly support newest silentpatch?

**Comment by CookiePLMonster** [comment]
The latest SP enables directional lighting on detached car parts. Might be that something in SkyGfx Extended doesn't play nice with this.

**Comment by MOnkeMan5009** [comment]
> The latest SP enables directional lighting on detached car parts. Might be that something in SkyGfx Extended doesn't play nice with this.

perhaps.. Could you please check ?

**Comment by CookiePLMonster** [comment]
I cannot reproduce with Extended v1.7 either. There might be more to it than just the existence of those two mods. ImVehFt perhaps?
![image](https://github.com/user-attachments/assets/0a10fc1d-a020-4c23-b4a8-a872ea6fbba8)

**Comment by MOnkeMan5009** [comment]
I have ImVehFt installed but not sure what i could do with it.  i mean it worked fine before, not sure. i know that i just replaced the newer SP with an older version and now the issue is gone

**Comment by CookiePLMonster** [comment]
> I have ImVehFt installed but not sure what i could do with it. i mean it worked fine before, not sure. i know that i just replaced the newer SP with an older version and now the issue is gone

Without the incompatibility nailed down precisely, I can't do anything about it.

**Comment by MOnkeMan5009** [comment]
> > I have ImVehFt installed but not sure what i could do with it. i mean it worked fine before, not sure. i know that i just replaced the newer SP with an older version and now the issue is gone
> 
> Without the incompatibility nailed down precisely, I can't do anything about it.

just tested right now as well, you're right. ImVehFt causes that thing only with latest silentpatch. the older one a bit, works just fine tho

**Comment by CookiePLMonster** [comment]
> just tested right now as well, you're right. ImVehFt causes that thing only with latest silentpatch. the older one a bit, works just fine tho

Yeah, that sounds more plausible. I'll look into it but it might be something out of my control and for ImVehFt to fix.

**Comment by MOnkeMan5009** [comment]
> > just tested right now as well, you're right. ImVehFt causes that thing only with latest silentpatch. the older one a bit, works just fine tho
> 
> Yeah, that sounds more plausible. I'll look into it but it might be something out of my control and for ImVehFt to fix.

Could a toggleable option be a thing at least as a workaround to not have to resort to older version of SP? Also if you can, to contact the dude behind ImVEhFt. im not familiar with who owns each mod

**Comment by CookiePLMonster** [comment]
That requires figuring out what happened first, and once that happens I might have a fix already.

**Comment by MOnkeMan5009** [comment]
> That requires figuring out what happened first, and once that happens I might have a fix already.

So, with these incompatibilities i noticed anything other than PC preset, made that bug occur.  which likely meant the car rendering thingy for PC too. So anything other like ps2 mobile etc, was broken with it

**Comment by CookiePLMonster** [comment]
I opened https://github.com/CookiePLMonster/SilentPatch/issues/111 to keep track of it on my side.

**Comment by RuesanG** [comment]
> I cannot reproduce with Extended v1.7 either. There might be more to it than just the existence of those two mods. ImVehFt perhaps? ![image](https://private-user-images.githubusercontent.com/7947461/384693051-0a10fc1d-a020-4c23-b4a8-a872ea6fbba8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MDY4NzIsIm5iZiI6MTczMTUwNjU3MiwicGF0aCI6Ii83OTQ3NDYxLzM4NDY5MzA1MS0wYTEwZmMxZC1hMDIwLTRjMjMtYjRhOC1hODcyZWE2ZmJiYTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMTQwMjUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDUwN2QyZTNjNGI2MDFlOWZjM2U0NWQzNDZmOWY1YzgwNjEyOGNlY2QyZjZjODg5M2I2NmUwZmFiMGY3YmY1NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GQk0ciXAzy3anjSDASNt6CjsqDgcT46cSREheqSLVyw)

For me it only happens when I use auto mods that include ImVehFT functions like "https://kiloryumods.my.canva.site/revampedvehiclesproject#downloads-2"

**Comment by Lethality5001** [comment]
> > I cannot reproduce with Extended v1.7 either. There might be more to it than just the existence of those two mods. ImVehFt perhaps? ![image](https://private-user-images.githubusercontent.com/7947461/384693051-0a10fc1d-a020-4c23-b4a8-a872ea6fbba8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MDY4NzIsIm5iZiI6MTczMTUwNjU3MiwicGF0aCI6Ii83OTQ3NDYxLzM4NDY5MzA1MS0wYTEwZmMxZC1hMDIwLTRjMjMtYjRhOC1hODcyZWE2ZmJiYTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMTQwMjUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDUwN2QyZTNjNGI2MDFlOWZjM2U0NWQzNDZmOWY1YzgwNjEyOGNlY2QyZjZjODg5M2I2NmUwZmFiMGY3YmY1NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GQk0ciXAzy3anjSDASNt6CjsqDgcT46cSREheqSLVyw)
> 
> For me it only happens when I use auto mods that include ImVehFT functions like "https://kiloryumods.my.canva.site/revampedvehiclesproject#downloads-2"

yeah, seems like silent said hes working on some compatibility fix

**Comment by CookiePLMonster** [comment]
> > I cannot reproduce with Extended v1.7 either. There might be more to it than just the existence of those two mods. ImVehFt perhaps? ![image](https://private-user-images.githubusercontent.com/7947461/384693051-0a10fc1d-a020-4c23-b4a8-a872ea6fbba8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MDY4NzIsIm5iZiI6MTczMTUwNjU3MiwicGF0aCI6Ii83OTQ3NDYxLzM4NDY5MzA1MS0wYTEwZmMxZC1hMDIwLTRjMjMtYjRhOC1hODcyZWE2ZmJiYTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMTQwMjUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDUwN2QyZTNjNGI2MDFlOWZjM2U0NWQzNDZmOWY1YzgwNjEyOGNlY2QyZjZjODg5M2I2NmUwZmFiMGY3YmY1NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GQk0ciXAzy3anjSDASNt6CjsqDgcT46cSREheqSLVyw)
> 
> For me it only happens when I use auto mods that include ImVehFT functions like "https://kiloryumods.my.canva.site/revampedvehiclesproject#downloads-2"


Is the "public" version enough to reproduce the issue?

**Comment by CookiePLMonster** [comment]
> > > > I cannot reproduce with Extended v1.7 either. There might be more to it than just the existence of those two mods. ImVehFt perhaps? ![image](https://private-user-images.githubusercontent.com/7947461/384693051-0a10fc1d-a020-4c23-b4a8-a872ea6fbba8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MDY4NzIsIm5iZiI6MTczMTUwNjU3MiwicGF0aCI6Ii83OTQ3NDYxLzM4NDY5MzA1MS0wYTEwZmMxZC1hMDIwLTRjMjMtYjRhOC1hODcyZWE2ZmJiYTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTNUMTQwMjUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDUwN2QyZTNjNGI2MDFlOWZjM2U0NWQzNDZmOWY1YzgwNjEyOGNlY2QyZjZjODg5M2I2NmUwZmFiMGY3YmY1NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GQk0ciXAzy3anjSDASNt6CjsqDgcT46cSREheqSLVyw)
> > > 
> > > 
> > > For me it only happens when I use auto mods that include ImVehFT functions like "https://kiloryumods.my.canva.site/revampedvehiclesproject#downloads-2"
> > 
> > 
> > Is the "public" version enough to reproduce the issue?
> 
> yeah.

Excellent, that'll be a good baseline for me to test with.
